### PR TITLE
Find max debt

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
can't be txn fees, unless they're negative
is this difference always 40? seems wonky. and this is with updating to a very recent block number (in last hour or so).

```
failed taking out FIAT debt = 92945.09522576001, decrement=39
failed taking out FIAT debt = 92944.09522576003, decrement=40
success taking out FIAT debt = 92943.09522576001, decrement=41
Current Fiat Balance:  92983.856566747101197346
lost in txn fees (?): -40.76134098708883 (-0.04385623363207239%)
```

**Comment by villanuevawill** [comment]
> can't be txn fees, unless they're negative is this difference always 40? seems wonky. and this is with updating to a very recent block number (in last hour or so).
> 
> ```
> scripts/simulate.js:158
> failed taking out FIAT debt = 92944.09522576003, decrement=40
> scripts/simulate.js:158
> success taking out FIAT debt = 92943.09522576001, decrement=41
> scripts/simulate.js:155
> Current Fiat Balance:  92983.856566747101197346
> scripts/simulate.js:164
> lost in txn fees (?): -40.76134098708883 (-0.04385623363207239%)```
> ```

Ah, no it's not always 40, I just hardcoded it temporarily. See discussion with fiat in element, we have to use a normalized value.

File: scripts/simulate.js [hunk 1]
```diff
@@ -145,12 +131,41 @@ async function main() {
...
+      'modifyCollateralAndDebt',
+      [
+        fiatDaiVaultAddress,
+        daiPTAddress,
+        0,
+        proxyAddress,
+        signer.address,
+        signer.address,
+        ptBalance,
+        tryDebt
```
Comment (ghost) [inline scripts/simulate.js:157]: I thought I took out the minus 40

**Comment by ghost** [comment]
increase in normalization rate is exact same as increase in normalized debt. actual debt taken out is the same as input, after adjusting for the change in normalization factor.
```
D:\nodejs\npx.cmd hardhat run ./scripts/simulate.js
confirmed transferred balance:  BigNumber {_hex: '0x152d02c7e14af6800000', _isBigNumber: true}
PTs Acquired:  100761.624685680483668325
Fair Price:  0.921272793229824671
Max Debt:  92828.943424552113537521
Virtual rate from publican: 0
Vault {totalNormalDebt: 9284.656645598245, rate: 1.000263018473884, debtCeiling: 3500000, debtFloor: 250, totalNormalDebtWAD: BigNumber, …}
Virtual rate from codex: 1.000263018473884
success taking out normalized FIAT debt = 92803.60608155816, decrement=0
Current Fiat Balance:  92828.845731606821505259
Current Fiat Normalized Balance:  92804.436450361965377381
difference in Normal Debt: input minus output: 0.8303688038156861 (0.0008947592005055677%)
difference in Actual Debt: input minus output: 0.8305872061574676 (0.0008947592005122908%)
new fair price:  0.921272798798487545
increase in price: 6.044531986049151e-7%
new normalization rate: 1.0002719684192714
increase in rate: 0.000894759200531503%
difference in Actual Debt using new normalization rate: 0 (0%)
```

**Comment by ghost** [comment]
tldr;
![image](https://user-images.githubusercontent.com/95365226/168454728-ddbbd28b-146b-48f0-b16a-9946a33229f2.png
